pkg/api/v1beta2/dynakube/feature_flags.go

```diff
@@ -0,0 +1,291 @@
+/*
+Copyright 2021 Dynatrace LLC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dynakube
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/logd"
+)
+
+const (
+	AnnotationFeaturePrefix = "feature.dynatrace.com/"
+
+	// General.
+	AnnotationFeaturePublicRegistry = AnnotationFeaturePrefix + "public-registry"
+
+	// activeGate.
+
+	// Deprecated: AnnotationFeatureDisableActiveGateUpdates use AnnotationFeatureActiveGateUpdates instead.
+	AnnotationFeatureDisableActiveGateUpdates = AnnotationFeaturePrefix + "disable-activegate-updates"
+
+	AnnotationFeatureActiveGateUpdates = AnnotationFeaturePrefix + "activegate-updates"
+
+	AnnotationFeatureActiveGateAppArmor                   = AnnotationFeaturePrefix + "activegate-apparmor"
+	AnnotationFeatureAutomaticK8sApiMonitoring            = AnnotationFeaturePrefix + "automatic-kubernetes-api-monitoring"
+	AnnotationFeatureAutomaticK8sApiMonitoringClusterName = AnnotationFeaturePrefix + "automatic-kubernetes-api-monitoring-cluster-name"
+	AnnotationFeatureK8sAppEnabled                        = AnnotationFeaturePrefix + "k8s-app-enabled"
+	AnnotationFeatureActiveGateIgnoreProxy                = AnnotationFeaturePrefix + "activegate-ignore-proxy"
+
+	// dtClient.
+
+	AnnotationFeatureNoProxy             = AnnotationFeaturePrefix + "no-proxy"
+	AnnotationFeatureApiRequestThreshold = AnnotationFeaturePrefix + "dynatrace-api-request-threshold"
```

Comment:
if we have a specific field for it then we shouldn't need it here, the conversion should handle it. (if it was present on the previous dynakube as a feature-flag, then it was moved to the specific field by the conversion)